cmd/metadata-server/server.go

```diff
@@ -214,3 +209,56 @@ func (s *Server) setMetadataFile(ctx context.Context, metadata *MetadataFile) er
 
 	return nil
 }
+
+func (s *Server) runGarbageCollection(ctx context.Context) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	metadata, err := s.getMetadataFile(ctx)
+	if err != nil {
+		return nil
+	}
+
+	keyMap := make(map[string]bool)
+	for k := range metadata.Entries {
+		keyMap[k] = true
+	}
+
+	fileKeysReq := &spb.GetFileKeysRequest{}
+	fileKeysRes, err := s.storageClient.GetFileKeys(ctx, fileKeysReq)
+	if err != nil {
+		return err
+	}
+
+	for key := range fileKeysRes.Keys {
+		actualKey := fileKeysRes.Keys[key]
+		if !keyMap[actualKey] {
+			deleteReq := &mpb.DeleteFileRequest{Key: actualKey}
+			if _, err := s.storageClient.DeleteFile(ctx, &spb.DeleteFileRequest{Key: deleteReq.Key}); err != nil {
+				return err
+			}
+
+			metadata, err := s.getMetadataFile(ctx)
```

Comment:
Isn't all this unnecessary? By definition these files aren't present in metadata, so why try to delete and reset metadata? It'll be unchanged.